presto-array/src/main/java/com/facebook/presto/array/SliceBigArray.java

```diff
@@ -61,22 +62,32 @@ public void set(long index, Slice value)
     {
         Slice currentValue = array.get(index);
         if (currentValue != null) {
-            sizeOfSlices -= getSize(currentValue);
+            int baseReferenceCount = trackedSlices.decrementReference(currentValue.getBase());
```

Comment:
as a side comment: `decrementReference` can better be named as `decrementReferenceAndGet` or simply `decrementAndGet` as    it's obvious from the name of the class (`ReferenceCountMap`) that we are manipulating references.

Reply:
ditto for `incrementReference`, a better name I think should be `incrementReferenceAndGet` or simply `incrementAndGet`.